ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -493,13 +494,50 @@ def _count_star(t, op):
     return sa.func.count(t.translate(ops.Where(where, 1, None)))
 
 
+def _count_distinct_star(t, op):
+    schema = op.arg.schema
+    cols = [sa.column(col, t.get_sqla_type(typ)) for col, typ in schema.items()]
+
+    if t._supports_tuple_syntax:
+        func = lambda *cols: sa.func.count(sa.distinct(sa.tuple_(*cols)))
+    else:
+        func = count_distinct
+
+    if (where := op.where) is None:
+        return func(*cols)
+
+    if t._has_reduction_filter_syntax:
+        return func(*cols).filter(t.translate(where))
+
+    if not t._supports_tuple_syntax and len(cols) > 1:
+        raise com.UnsupportedOperationError(
+            f"{t._dialect_name} backend doesn't support `COUNT(DISTINCT ...)` with a "
+            "filter with more than one column"
+        )
+
+    return sa.func.count(t.translate(ops.Where(where, sa.distinct(*cols), None)))
```

Comment:
```suggestion
    if op.where is None:
        return func(*cols)

    if t._has_reduction_filter_syntax:
        return func(*cols).filter(t.translate(op.where))

    if not t._supports_tuple_syntax and len(cols) > 1:
        raise com.UnsupportedOperationError(
            f"{t._dialect_name} backend doesn't support `COUNT(DISTINCT ...)` with a "
            "filter with more than one column"
        )

    return sa.func.count(t.translate(ops.Where(op.where, sa.distinct(*cols), None)))
```

Reply:
I prefer direct `op.` attribute references too, it shouldn't have a performance impact due to the slotted access, but helps with code readability.